### PR TITLE
Query Library: Enable edit description

### LINKED ISSUE #84895: Explore: Edit query template description
* Link to edit description in the kebab menu
* It should reuse the same modal that is used for saving. For now it will allow to change only the description but we can add more stuff in there later.

### PR DESCRIPTION
**Pull request by gelicia**
**What is this feature?**

Adds the ability to edit the description of a query template.

Frontend platform - there is a small edit to the fetch core utility. The Kubernetes API has a variety of ways to do `PATCH` requests, but knowing which one is being applied relies on the `content-type` header. Without adding the new logic to the fetch, it doesn't parse the JSON and won't work. Let me know if some other approach should be done here.

This removes the AddToLibraryForm and replaces it with a more generic QueryTemplateForm to handle adds and edits.\

We got a request to move the update logic to use a global type but I am opting to do that in a different ticket - #90024 

**Which issue(s) does this PR fix?**:

Fixes #84895

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/query-library/api/query.ts [hunk 1]
```diff
@@ -41,6 +42,9 @@ export const baseQuery: BaseQueryFn<QueryLibraryBackendRequest, DataQuerySpecRes
       showErrorAlert: true,
       method: requestOptions.method || 'GET',
       data: requestOptions.data,
+      headers: {
+        'Content-type': 'application/json-patch+json',
```
Comment (gelicia) [inline public/app/features/query-library/api/query.ts:46]: todo: not everything is a json patch

File: public/app/core/utils/fetch.ts [hunk 2]
```diff
@@ -74,7 +74,10 @@ export const isContentTypeApplicationJson = (headers: Headers) => {
   }
 
   const contentType = headers.get('content-type');
-  if (contentType && contentType.toLowerCase() === 'application/json') {
+  if (
+    contentType &&
+    (contentType.toLowerCase() === 'application/json' || contentType.toLowerCase() === 'application/json-patch+json')
```
Comment (gelicia) [inline public/app/core/utils/fetch.ts:79]: Extending what counts as true for this function will be necessary for PATCH operations in the API platform, which require this content type to be set.

**Review by harisrozajac — APPROVED** [review]
great work! 🚀 tested updating ql entries descriptions, verified that the content-type when sending the request is `application/merge-patch+json`

File: public/app/core/utils/fetch.ts [hunk 3]
```diff
@@ -74,7 +74,10 @@ export const isContentTypeApplicationJson = (headers: Headers) => {
   }
 
   const contentType = headers.get('content-type');
-  if (contentType && contentType.toLowerCase() === 'application/json') {
+  if (
+    contentType &&
+    (contentType.toLowerCase() === 'application/json' || contentType.toLowerCase() === 'application/merge-patch+json')
+  ) {
```
Comment (ashharrison90) [inline public/app/core/utils/fetch.ts:80]: this question might be very obvious... is it still appropriate to include this logic in a function that's called `isContentTypeApplicationJson`? 🤔 or should we:
- have a separate function (`isContentTypeMergePatchJson`?) and check both of them in `backend_srv`?
- rename the function to be less specific?
- leave it as-is?

idk enough about the difference between `application/json` and `application/merge-patch+json` to have an opinion tbh 😅 just raising it

Comment (gelicia) [reply public/app/core/utils/fetch.ts:80]: @ashharrison90 - The context behind this content type is it's to work with the API platform. The Kubernetes API it's built off of has several options for doing patches, which are differentiated by the content type https://kubernetes.io/docs/reference/using-api/api-concepts/#patch-and-apply

The merge patch kind used here is the one that is most intuitive to most people when they think of patching an update - just a JSON object with the values you'd want to change. There will be no difference between `application/json` and `application/merge-patch+json` in what the content actually is - its all just JSON - but we need the specific content type for the patch to apply as needed.

I could go with separating this function, or renaming it to `isContentTypeJson` to be more generic. Whatever you prefer!

Comment (ashharrison90) [reply public/app/core/utils/fetch.ts:80]: i'd be fine with leaving the logic as-is and just renaming the function `isContentTypeJson` 👍 

but honestly i don't think my opinion should hold much weight, i'm sure you/app platform know best 🙌

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. This is a friendly reminder that the Steady [11.2.x Release Branch](https://github.com/grafana/grafana/tree/v11.2.x) was cut this week, and that it includes the changes committed up to https://github.com/grafana/grafana/commit/b4126d3bce3627e9460e2db0e870c8d984be8e84. This PR was merged after it, so we wanted to let you know that you can add the `backport v11.2.x` label to to backport it and make it part of the next 11.2.x Patch Release, just in case you may have missed the news.

Thank you!

**Pull request merged**